src/main/java/it/pn/frontend/e2e/pages/destinatario/personaFisica/ITuoiRecapitiPage.java

```diff
@@ -38,4 +61,172 @@ public void waitLoadITuoiRecapitiPage() {
+            inserimentoPecField.sendKeys(emailPEC);
+        }
+    }
+    public void confermaButtonClick() {
+        this.confermaButton.click();
+    }
+
+    public void waitLoadPopUp() {
+        try {
+            By titleBy = By.xpath("//h2[contains(@id,'dialog-title')]");
+            this.getWebDriverWait(30).until(ExpectedConditions.visibilityOfElementLocated(titleBy));
+            logger.info("Il po-up di conferma viene visualizzato correttamente");
+        }catch (TimeoutException e){
+            logger.error("Il pop-up di conferma NON viene visualizzato correttamente con errori: "+e.getMessage());
+            Assert.fail("Il pop-up di conferma NON viene visualizzato correttamente con errori:"+e.getMessage());
+        }
+    }
+
+    public void sendOTP(String otp) {
+        String[] otps = otp.split("");
+        try{
+            By otpInputby = By.xpath("//input[contains(@id,'code-input')]");
+            getWebDriverWait(30).until(ExpectedConditions.visibilityOfElementLocated(otpInputby));
+            List<WebElement> otpInputs= this.elements(otpInputby);
+            for (int i=0; i<otps.length; i++) {
+                otpInputs.get(i).sendKeys(otps[i]);
+            }
+            logger.info("Il codice otp viene inserito correttamente");
+        }catch (TimeoutException e){
+            logger.error("Il codice otp NON viene inserito correttamente con errore:"+e.getMessage());
+            Assert.fail("Il codice otp NON viene inserito correttamente con errore:"+e.getMessage());
+        }
+    }
+
+    public void waitMessaggioErrore() {
+        try {
+            By messaggioErroreBy = By.xpath("//div[contains(@data-testid, 'errorAlert')]");
+            this.getWebDriverWait(30).until(ExpectedConditions.visibilityOfElementLocated(messaggioErroreBy));
+            logger.info("Il messaggio di errore viene visualizzato correttamente");
+        }catch (TimeoutException e){
+            logger.error("Il messaggio di errore non viene visualizzato con errore: "+e.getMessage());
+            Assert.fail("Il messaggio di errore non viene visualizzato con errore: "+e.getMessage());
+        }
+    }
+
+    public void annullaButtonClick() {
+        this.annullaButton.click();
+    }
+
+    public void confermaButtonClickPopUp() {
+        this.confermaButtonPopUp.click();
+    }
+
+    public String getPecErrorMessage(){
+        By errorBy = By.id("pec-helper-text");
+        WebElement errorMessage = driver.findElement(errorBy);
+        this.getWebDriverWait(30).until(ExpectedConditions.visibilityOf(errorMessage));
+        return errorMessage.getText();
+    }
+
+    public boolean verificaButtoneConfermaDisabilitato(){
+        try{
+            getWebDriverWait(30).until(ExpectedConditions.visibilityOf(this.confermaButton));
+            return Boolean.parseBoolean(this.confermaButton.getAttribute("disabled"));
+        }catch (NoSuchElementException | TimeoutException e){
+            return false;
```

Comment:
insert log

Reply:
Logger already present in the step that calls this method